module/common.nf

```diff
@@ -72,22 +72,29 @@ process rename_samples_BCFtools {
         saveAs: { "${task.process.split(':')[-1]}-${var_type}/log${file(it).getName()}" }
 
     input:
-    val normal_id 
-    val tumor_id 
+    val ids
     tuple val(var_type), path(vcf)
 
     output:
-    tuple val(var_type), path("*.vcf.gz"), emit: gzvcf
+    tuple val(var_type), path("*.vcf"), emit: vcf
     path ".command.*"
     path "*_samples.txt"
 
     script:
+    rename_lines = ids
+        .collect { "${it['orig_id']}\t${it['id']}" }
+        .join("\n")
+    sed_replacements = ids
+        .collect { "sed 's/ample=${it['orig_id']}/ample=${it['id']}/g'" }
+        .join(" | ")
     """
     set -euo pipefail
-    echo -e 'NORMAL\t${normal_id}' > ${params.output_filename}_samples.txt
-    echo -e 'TUMOR\t${tumor_id}' >> ${params.output_filename}_samples.txt
-    bcftools reheader -s ${params.output_filename}_samples.txt --output ${params.output_filename}_${var_type}.vcf.gz ${vcf}
-    """
+    echo -e '${rename_lines}' > ${params.output_filename}_samples.txt
+    bcftools reheader -s ${params.output_filename}_samples.txt ${vcf} \
+        | gunzip -c \
```

Comment:
Would the bcftools common options allow for outputting the file directly in raw text format rather than zipping and unzipping?